std/special/compiler_rt/arm/aeabi_fcmp.zig

```diff
@@ -0,0 +1,108 @@
+// Ported from:
+//
+// https://github.com/llvm/llvm-project/commit/d674d96bc56c0f377879d01c9d8dfdaaa7859cdb/compiler-rt/lib/builtins/arm/aeabi_fcmp.S
+
+const compiler_rt_armhf_target = false; // TODO
```

Comment:
This should be tied to `builtin` state and become a compile error if not possible.

Reply:
That would disable every hard floating point arm32 target because there is no support for hard floating point at all currently and release-fast/release-safe mode include even unused built-ins #2062. Such a change is beyond my area of responsibility. On a different compiler-rt pull request I did I just didn't add the assembly section at all. This time I marked it with a TODO, I'm not sure why I marked it this time.

Because the port of compiler-rt is so large, I think it might make more sense to incrementally add hard floating point support (my current plan), rather than just say it won't work at all. I am working my way through the soft floating point routines I need to work around #2062 first.

I know we have #1290, but maybe the hard floating point support on arm32 targets could be it's own issue so it doesn't get lost in the sauce. It also would document the plan to add it explicitly.

Can i request the opinion of @andrewrk?

Reply:
Even without TODOs in the code, its very easy to see what needs to be added from compiler-rt to support hard floating point. Just grep for `COMPILER_RT_ARMHF_TARGET`.

Reply:
The reason I think that should be a compiler error is that if the floatABI is set to hard and we don't deliver hard, then it seems like a fallacy. The `@compileError` should mention to set the floatABI to `soft`.

Reply:
Would you re-enable the hard float abi once every function was complete? That is a lot of code to wait on.

It would also make testing the addition of each function more difficult as you would have to go through and comment out the compile errors in order to get a binary with the hard float routine you are working on in it.

Reply:
Whether or not hard float is desired is available via this logic:

```zig
const compiler_rt_armhf_target = switch (builtin.abi) { .gnueabihf, .eabihf, .musleabihf => true, else => false };
```

I get that compiler-rt is incomplete, especially for these targets, but aren't compile errors better than silently incorrect ABI?